test/unit/Hbar.js

```diff
@@ -2,20 +2,169 @@ import Hbar from "../src/Hbar.js";
+            "115100000"
+        );
+        expect(Hbar.fromString("1 tℏ").toTinybars().toString()).to.be.equal(
+            "1"
+        );
+        expect(Hbar.fromString("1 μℏ").toTinybars().toString()).to.be.equal(
+            "100"
+        );
+        expect(Hbar.fromString("1 mℏ").toTinybars().toString()).to.be.equal(
+            "100000"
+        );
+        expect(Hbar.fromString("1 ℏ").toTinybars().toString()).to.be.equal(
+            "100000000"
+        );
+        expect(Hbar.fromString("1 kℏ").toTinybars().toString()).to.be.equal(
+            "100000000000"
+        );
+        expect(Hbar.fromString("1 Mℏ").toTinybars().toString()).to.be.equal(
+            "100000000000000"
+        );
+        expect(Hbar.fromString("1 Gℏ").toTinybars().toString()).to.be.equal(
+            "100000000000000000"
+        );
+        expect(Hbar.fromString("1.151 mℏ").toTinybars().toString()).to.be.equal(
+            "115100"
+        );
+        expect(Hbar.fromString("1.151 ℏ").toTinybars().toString()).to.be.equal(
+            "115100000"
+        );
+        expect(Hbar.fromString("1.151 kℏ").toTinybars().toString()).to.be.equal(
+            "115100000000"
+        );
+        expect(Hbar.fromString("1.151 Mℏ").toTinybars().toString()).to.be.equal(
+            "115100000000000"
+        );
+        expect(Hbar.fromString("1.151 Gℏ").toTinybars().toString()).to.be.equal(
+            "115100000000000000"
+        );
+        expect(Hbar.fromString("+1").toTinybars().toString()).to.be.equal(
+            "100000000"
+        );
+        expect(Hbar.fromString("+1.151").toTinybars().toString()).to.be.equal(
+            "115100000"
+        );
+        expect(Hbar.fromString("+1 Gℏ").toTinybars().toString()).to.be.equal(
+            "100000000000000000"
+        );
+        expect(
+            Hbar.fromString("+1.151 Gℏ").toTinybars().toString()
+        ).to.be.equal("115100000000000000");
+        expect(Hbar.fromString("-1").toTinybars().toString()).to.be.equal(
+            "-100000000"
+        );
+        expect(Hbar.fromString("-1.151").toTinybars().toString()).to.be.equal(
+            "-115100000"
+        );
+        expect(Hbar.fromString("-1 Gℏ").toTinybars().toString()).to.be.equal(
+            "-100000000000000000"
+        );
         expect(
-            Hbar.fromTinybars(1234567890).toString(HbarUnit.Microbar)
-        ).to.eql("12345678.9 μℏ");
+            Hbar.fromString("-1.151 Gℏ").toTinybars().toString()
+        ).to.be.equal("-115100000000000000");
     });
 
-    it("should convert from hbars to tinybars", function () {
-        const tinybar = "98327923153";
-        const hbar = new Hbar("983.27923153");
+    it('should throw error "Hbar in tinybars contains decimals"', function () {
+        let check = ["1.151 tℏ", "1.151 μℏ", "1.151", "-1.151", "+1.151"];
 
-        expect(tinybar).to.eql(hbar.toTinybars().toString());
+        for (let i = 0; i < check.length; i++) {
```

Comment:
Should use a for each loop here instead. Also, isn't this test the same as the one above where you're using `i`, `j`, and `k`?

Reply:
I'll look into for each. The tests are not the same. This one tests the specific "Hbar in tinybars contains decimals". The i,j,k tests the regex.